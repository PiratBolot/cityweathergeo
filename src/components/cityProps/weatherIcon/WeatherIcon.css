.weather_icon {
    max-width: 40px;
    max-height: 40px;
    filter: drop-shadow(2px 4px 6px black);
    padding: 10px 20px 5px 20px;
}

.weather_frame {
    padding: 3px;
}