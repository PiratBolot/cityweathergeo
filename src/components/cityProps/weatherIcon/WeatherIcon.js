import React from 'react';
import './WeatherIcon.css';

class WeatherIcon extends React.Component {
    render = () => (
        <img className="weather_icon" src={"http://openweathermap.org/img/wn/" +
        this.props.icon + "@2x.png"} alt=""
        />
    )
}

export default WeatherIcon;