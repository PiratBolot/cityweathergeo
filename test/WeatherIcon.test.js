import React from 'react';
import { shallow } from '../enzyme/enzyme';
import { shallowToJson } from 'enzyme-to-json';

import WeatherIcon from "../src/components/cityProps/weatherIcon/WeatherIcon";

describe('Weather Icon', () => {
    const iconCode = "04n";

    it('should render correctly', () => {
        const output = shallow(
            <WeatherIcon icon={iconCode}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});