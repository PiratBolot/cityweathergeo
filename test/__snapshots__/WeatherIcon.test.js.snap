// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Icon should render correctly 1`] = `
<img
  alt=""
  className="weather_icon"
  src="http://openweathermap.org/img/wn/04n@2x.png"
/>
`;
